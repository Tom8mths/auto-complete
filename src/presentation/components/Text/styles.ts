import { TextProps } from '.';

type TextStyleProps = Pick<TextProps, 'color' | 'size'>;

export const sizeModiffiers = {
  xsmall: 'text-xs',
  small: 'text-sm',
  base: 'text-base',
  large: 'text-lg',
  xlarge5: 'text-5xl',
};

export const colorModiffiers = {
  white: 'text-white',
  black: 'text-black',
  lightGray: 'text-neutral-400',
  purple: 'text-purple-500',
};

export const text = ({ color, size }: TextStyleProps) => `
  ${!!size && sizeModiffiers[size]}
  ${!!color && colorModiffiers[color]}
`;
