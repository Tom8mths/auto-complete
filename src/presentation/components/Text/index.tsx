import * as S from './styles';

export type TextProps = {
  children: string | React.ReactNode;
  size?: 'xsmall' | 'small' | 'base' | 'large' | 'xlarge5';
  color?: 'white' | 'lightGray' | 'black' | 'purple';
  className?: string;
};

export default function Text({
  size = 'base',
  color = 'white',
  children,
  className = '',
}: TextProps) {
  return (
    <p className={`${S.text({ size, color })} ${className}`}>{children}</p>
  );
}
