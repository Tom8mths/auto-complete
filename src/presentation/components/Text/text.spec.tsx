import { render, screen } from "@testing-library/react";

import Text from ".";

describe("<Text />", () => {
  it("should render the heading", () => {
    const { container } = render(<Text>My awesome text</Text>);

    const headingElement = screen.getByText(/my awesome text/i);

    expect(headingElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
