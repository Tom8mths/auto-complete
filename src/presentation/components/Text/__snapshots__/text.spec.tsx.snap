// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render the heading 1`] = `
<div>
  <p
    class="
  text-base
  text-white
 "
  >
    My awesome text
  </p>
</div>
`;
